packages/react-accordion/src/components/AccordionHeader/useAccordionHeader.ts

```diff
@@ -35,58 +35,49 @@ export const accordionHeaderShorthandProps: Array<keyof AccordionHeaderSlots> =
  */
 export const useAccordionHeader = (props: AccordionHeaderProps, ref: React.Ref<HTMLElement>): AccordionHeaderState => {
   const { onHeaderClick: onAccordionHeaderClick, disabled, open } = useAccordionItemContext();
-  const button = useContextSelector(AccordionContext, ctx => ctx.button);
-  const expandIcon = useContextSelector(AccordionContext, ctx => ctx.expandIcon);
-  const inline = useContextSelector(AccordionContext, ctx => ctx.inline);
-  const icon = useContextSelector(AccordionContext, ctx => ctx.icon);
-  const expandIconPosition = useContextSelector(AccordionContext, ctx => ctx.expandIconPosition);
-  const size = useContextSelector(AccordionContext, ctx => ctx.size);
   const id = useId('accordion-header-', props.id);
   const panel = useDescendants(accordionItemDescendantContext)[1] as AccordionItemDescendant | undefined;
   const innerRef = React.useRef<HTMLElement>(null);
-  const initialState = {
+
+  useAccordionItemDescendant(
+    {
+      element: innerRef.current,
+      id,
+    },
+    0,
+  );
+
+  const originalHandleClick = isObjectShorthandProps(props.button) ? props.button.onClick : undefined;
+
+  return {
     ...props,
+    disabled,
+    open,
     ref: useMergedRefs(ref, innerRef),
-    size: props.size ?? size ?? 'medium',
-    inline: props.inline ?? inline ?? false,
-    role: props.role ?? 'heading',
-    expandIconPosition: props.expandIconPosition ?? expandIconPosition ?? 'start',
+    size: props.size || 'medium',
+    inline: props.inline || false,
```

Comment:
I am not fully sure that it's a good idea to remove these props from context. It makes things simpler for us, but complicates usage for customers. Can you please clarify reasoning behind of it?

Reply:
The reason is the same for removing slots: If reusability is the reason than composition should be used instead of hoisting state. The user would only have to make a `InlineAccordionHeader` or something like that

Reply:
Agree, I like removing slots from context, but props and such do make sense to be propagated

Reply:
Oh boy. I don't know, I'll give this a thought and perhaps reintroduce then in a different PR